### PR TITLE
Don't push to cache buster if attachment path is nil

### PR DESCRIPTION
**Pull request by tribela**
Old remote media which already deleted by scheduler was causing an error by passing `nil` value into `CacheBuster`

Related: #31353

### CODE REVIEW

File: spec/models/media_attachment_spec.rb
Comment (ClearlyClaire) [inline spec/models/media_attachment_spec.rb:1]: I wouldn't change the spec, as those files are expected to exist, and if that ever changes, this is unexpected and may silence failures.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Hi! Thanks for your contribution!

The application code change is good, but I wouldn't change the spec here.

File: spec/models/media_attachment_spec.rb
Comment (ClearlyClaire) [reply spec/models/media_attachment_spec.rb:1]: Oh, right, `thumbnail_path` is `nil` in this test… hm, I guess I'd remove the check on `thumbnail_path` then (and possibly have another test for other media attachment types that use a thumbnail).

Comment (ClearlyClaire) [reply spec/models/media_attachment_spec.rb:1]: Yes, can you use the previous version, but without `thumbnail_path` (which is `nil` in the test)?

**Review by ClearlyClaire — APPROVED** [review]

**Comment by tribela** [comment]
Oh.. build is failed at asset compiling. I think it is not related with this change

**Pull request merged**